### PR TITLE
Move event utils to a separate file

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: packages/core/src/eventUtils.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * @param delayRef The delay in milliseconds
+ * @param id The state node ID where this event is handled
+ */
+export function after(delayRef: number | string, id?: string) {
+  const idSuffix = id ? `#${id}` : '';
+  return `${constantPrefixes.after}(${delayRef})${idSuffix}`;
+}
+
+export function doneStateEventType<T extends string = string>(id: T) {
```
Comment (Andarist) [inline packages/core/src/eventUtils.ts:21]: let's bikeshed how those helper functions should be named (how to call the utils responsible for creating the event type and how do we call utils responsible for creating the event itself). We also need to decide which ones should be publicly exported (if any)

File: packages/core/src/constantPrefixes.ts [hunk 2]
```diff
@@ -1,7 +1,6 @@
...
-export const doneState = ConstantPrefix.DoneState;
-export const errorExecution = ConstantPrefix.ErrorExecution;
-export const errorPlatform = ConstantPrefix.ErrorPlatform;
-export const error = ConstantPrefix.ErrorCustom;
+export const after = 'xstate.after';
+export const doneInvoke = 'done.invoke';
+export const doneState = 'done.state';
+export const error = 'xstate.error';
+export const errorExecution = 'error.execution';
+export const errorPlatform = 'error.platform';
```
Comment (Andarist) [inline packages/core/src/constantPrefixes.ts:6]: I think I would take a step further and just remove this file altogether. Inlining those where needed should be fine - it's not like we are using this all over the place

File: packages/core/src/eventUtils.ts (hunk 1 above)
Comment (davidkpiano) [reply packages/core/src/eventUtils.ts:21]: By default, let's not export any

**Review by davidkpiano — APPROVED** [review]

File: packages/core/src/constantPrefixes.ts (hunk 2 above)
Comment (davidkpiano) [reply packages/core/src/constantPrefixes.ts:6]: Agree

File: packages/core/src/eventUtils.ts (hunk 1 above)
Comment (Andarist) [reply packages/core/src/eventUtils.ts]: I pushed out extra changes, could you re-review?

**Review by davidkpiano — APPROVED** [review]

**Pull request merged**